libqtile/widget/tasklist.py

```diff
@@ -288,10 +304,19 @@ def calc_box_widths(self):
     def _configure(self, qtile, bar):
         base._Widget._configure(self, qtile, bar)
 
-        if qtile.core.name == "wayland" and self.icon_size != 0:
+        if not has_xdg and self.theme_mode is not None:
+            logger.warning("You must install pyxdg to use theme icons.")
+            self.theme_mode = None
+
+        if self.theme_mode not in ["preferred", "fallback"]:
```

Comment:
`None` seems to be a valid value so in that case the warning may be unnecessary?

```suggestion
        if self.theme_mode and self.theme_mode not in ["preferred", "fallback"]:
```

Reply:
Agree with the change but not sure I follow your comment. If `theme_mode` is `None` then we don't use xdg icons. So `None` is valid but it disables icons. If someone's entered a string here then they want icons but have made a mistake with the value.

Reply:
I was just thinking about the warning message. If it was `None` then `"Unexpected theme_mode None. Theme icons will be disabled."` would have been logged, which may have been confusing to users because it isn't unexpected.

LGTM!

Reply:
Ah, of course.